ocs_deploy/waf.py

```diff
@@ -1,27 +1,102 @@
-﻿# ocs_deploy/stacks/waf.py
-from aws_cdk import Stack, RemovalPolicy, aws_wafv2 as wafv2, aws_logs as logs, aws_iam as iam, CfnOutput
+# ocs_deploy/stacks/waf.py
+from aws_cdk import (
+    Stack,
+    RemovalPolicy,
+    aws_wafv2 as wafv2,
+    aws_logs as logs,
+    aws_iam as iam,
+    CfnOutput,
+)
 from constructs import Construct
 from ocs_deploy.config import OCSConfig
 
+# URI patterns for endpoints that can send large POST bodies
+# These bypass only SizeRestrictions_BODY, all other protections remain active
+SizeRestrictions_BODY = [
+    r"^a/([-a-zA-Z0-9_]+)/assistants/new/$",
+    r"^a/([-a-zA-Z0-9_]+)/documents/collections/([0-9]+)/add_files$",
+    r"^a/([-a-zA-Z0-9_]+)/evaluations/dataset/new/$",
+    r"^a/([-a-zA-Z0-9_]+)/evaluations/evaluator/new/$",
+    r"^a/([-a-zA-Z0-9_]+)/evaluations/parse_csv_columns/$",
+    r"^a/([-a-zA-Z0-9_]+)/pipelines/data/([0-9]+)/$",
+    r"^slack/events$",
+    r"^users/profile/upload\-image/$",
+]
+
+# URI patterns for endpoints that may not send User-Agent header
+# These bypass only NoUserAgent_HEADER, all other protections remain active
+NoUserAgent_HEADER = [
+    r"^a/([-a-zA-Z0-9_]+)/chatbots/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/s/([^/]+)/chat/$",
+    r"^a/([-a-zA-Z0-9_]+)/chatbots/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/start/$",
+    r"^channels/sureadhere/([^/]+)/incoming_message$",
+    r"^channels/telegram/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})$",
+]
+
+
 class WAFStack(Stack):
     """
     Represents a CDK stack for deploying a WAF Web ACL associated with an Application Load Balancer.
-    Includes AWS Managed Rules and rate limiting in count mode, with logging to CloudWatch.
+
+    Protection Strategy:
+    1. IP blocklists (highest priority) - block malicious IPs
+    2. Scope-down allow rules - allow specific conditions to bypass ONLY specific checks:
+       - Large body (>8KB) for specific paths: bypass SizeRestrictions_BODY only when body is actually large
+       - Missing User-Agent for specific paths: bypass NoUserAgent_HEADER only when header is actually missing
+    3. AWS Managed Rules Common Rule Set - applies to all other traffic
+    4. Rate limiting (COUNT mode) - 2000 req/IP/5min
```

Comment:
2000 req/IP/5min feels like a lot?

Reply:
It is high - this is just what got added originally by the contractor how wrote this module. We can certainly change it. I don't think it's been triggered yet.